runtime/ast/unmarshal.go

```diff
@@ -0,0 +1,901 @@
+		Declarations: &a.declarations,
+	})
+}
+
+func (a *NominalType) UnmarshalJSON(b []byte) error {
+	type Alias NominalType
+	return json.Unmarshal(b, &struct {
+		*Alias
+	}{
+		Alias: (*Alias)(a)})
+}
+
+func (a *Operation) UnmarshalJSON(b []byte) error {
+	if op, ok := operationMap[strings.Trim(string(b), `"`)]; ok {
+		*a = op
+		return nil
+	}
+	return fmt.Errorf("unknown Operation %s", string(b))
+}
+
+func (a *OptionalType) UnmarshalJSON(b []byte) error {
+	type Alias OptionalType
+	return json.Unmarshal(b, &struct {
+		*Alias
+		ElementType TypeUnmarshaler
+		EndPos      *Position
+	}{
+		Alias:       (*Alias)(a),
+		ElementType: TypeUnmarshaler{&a.Type},
+		EndPos:      &a.EndPos,
+	})
+}
+
+func (a *PragmaDeclaration) UnmarshalJSON(b []byte) error {
+	type Alias PragmaDeclaration
+	return json.Unmarshal(b, &struct {
+		*Alias
+		Expression ExpressionUnmarshaler
+	}{
+		Alias:      (*Alias)(a),
+		Expression: ExpressionUnmarshaler{&a.Expression},
+	})
+}
+
+func (a *Program) UnmarshalJSON(b []byte) error {
+	type Alias Program
+	return json.Unmarshal(b, &struct {
+		*Alias
+		Declarations *[]Declaration
+	}{
+		Alias:        (*Alias)(a),
+		Declarations: &a.declarations,
+	})
+}
+
+func (a *ReferenceType) UnmarshalJSON(b []byte) error {
+	type Alias ReferenceType
+	return json.Unmarshal(b, &struct {
+		*Alias
+		ReferencedType TypeUnmarshaler
+		StartPos       *Position
+	}{
+		Alias:          (*Alias)(a),
+		ReferencedType: TypeUnmarshaler{&a.Type},
+		StartPos:       &a.StartPos,
+	})
+}
+
+func (a *RestrictedType) UnmarshalJSON(b []byte) error {
+	type Alias RestrictedType
+	return json.Unmarshal(b, &struct {
+		*Alias
+		RestrictedType TypeUnmarshaler
+	}{
+		Alias:          (*Alias)(a),
+		RestrictedType: TypeUnmarshaler{&a.Type},
+	})
+}
+
+func (a *ReturnStatement) UnmarshalJSON(b []byte) error {
+	type Alias ReturnStatement
+	return json.Unmarshal(b, &struct {
+		*Alias
+		Expression ExpressionUnmarshaler
+	}{
+		Alias:      (*Alias)(a),
+		Expression: ExpressionUnmarshaler{&a.Expression},
+	})
+}
+
+func (a *SwapStatement) UnmarshalJSON(b []byte) error {
+	type Alias SwapStatement
+	return json.Unmarshal(b, &struct {
+		*Alias
+		Left  ExpressionUnmarshaler
+		Right ExpressionUnmarshaler
+	}{
+		Alias: (*Alias)(a),
+		Left:  ExpressionUnmarshaler{&a.Left},
+		Right: ExpressionUnmarshaler{&a.Right},
+	})
+}
+
+func (a *SwitchStatement) UnmarshalJSON(b []byte) error {
+	type Alias SwitchStatement
+	return json.Unmarshal(b, &struct {
+		*Alias
+		Expression ExpressionUnmarshaler
+	}{
+		Alias:      (*Alias)(a),
+		Expression: ExpressionUnmarshaler{&a.Expression},
+	})
+}
+
+func (a *Transfer) UnmarshalJSON(b []byte) error {
+	type Alias Transfer
+	return json.Unmarshal(b, &struct {
+		*Alias
+		StartPos *Position
+	}{
+		Alias:    (*Alias)(a),
+		StartPos: &a.Pos,
+	})
+}
+
+func (a *TransferOperation) UnmarshalJSON(b []byte) error {
+	if op, ok := transferOperationMap[strings.Trim(string(b), `"`)]; ok {
```

Comment:
Instead of trimming the serialized string's quotes manually, potentially leaving quoted content, unmarshal as a string:

```go
var s string
if err := json.Unmarshal(b, &s); err != nil {
	return err
}
```